osf/models/mixins.py

```diff
@@ -1961,11 +1961,43 @@ def log_params(self):
     def get_spam_fields(self):
         return NotImplementedError()
 
-    def confirm_spam(self, save=False):
-        super(SpamOverrideMixin, self).confirm_spam(save=False)
+    def confirm_spam(self, save=True):
+        """
+        This should add behavior specific nodes/preprints confirmed to be spam.
+        :param save:
+        :return:
+        """
+        super().confirm_spam(save=save)
+        self.deleted = timezone.now()
         self.set_privacy('private', auth=None, log=False, save=False)
+
+        log = self.add_log(
+            action=self.log_class.CONFIRM_SPAM,
+            params=self.log_params,
+            auth=None,
+            save=False
+        )
+        log.should_hide = True
+        log.save()
+        if save:
+            self.save()
+
+    def confirm_ham(self, save=False):
+        """
+        This should add behavior specific nodes/preprints confirmed to be ham.
+        :param save:
+        :return:
+        """
+        super().confirm_ham()
+
+        if self.logs.filter(action=self.log_class.CONFIRM_SPAM):
```

Comment:
Shouldn't this be `FLAG_SPAM`?

Reply:
I don't think so, flagging something as spam shouldn't do anything, but make it show up in a list for an admin to see.

Reply:
https://github.com/CenterForOpenScience/osf.io/pull/9411/files#diff-e5c42b2bf5452803ba79e98c6dad492bR2091-R2092

Reply:
I still think `CONFIRM_SPAM` is correct with the current workflow though.

Reply:
We are fixing this for autoban, which will flag spam and make it private, yes? If something is autobanned, which causes it to become private, but was never confirmed as spam by an admin, will this confirmation as ham make it become public?

Reply:
My understanding is that the new autoban just takes things which are automatically flagged as spam and then confirm them as spam without admin approval. So what get's flagged and when it's confirmed isn't really a part of this PR. This is just the confirming process, I think.

Reply:
`SPAM_ACCOUNT_SUSPENSION_ENABLED` and `SPAM_FLAGGED_MAKE_NODE_PRIVATE` are separate features. The former bans a newly-created user if their content is flagged as spam. The latter makes things private when flagged as spam, regardless of recency.

https://github.com/CenterForOpenScience/osf.io/blob/20.10.0/osf/models/mixins.py#L2024-L2028

https://github.com/CenterForOpenScience/osf.io/blob/20.10.0/osf/models/mixins.py#L2055-L2060

Reply:
In theory, `SPAM_FLAGGED_MAKE_NODE_PRIVATE` should only cause things that are public to be made private. This isn't necessarily correct, due to over-checking of preprints, but that should be handled by the type/state checking below.

Reply:
So with `SPAM_FLAGGED_MAKE_NODE_PRIVATE` set to true, nodes that are not confirmed but are flagged will be set to private? But are there situations where this would produce the wrong behavior? Are private nodes ever flagged as spam? Should https://github.com/CenterForOpenScience/osf.io/blob/20.10.0/osf/models/mixins.py#L2055-L2060 confirm the spam when it marks it as private, so we have a consistent state?

Reply:
Didn't we decide on the Zoom that `FLAG_SPAM` was more appropriate here, because `FLAG_SPAM` only gets put on nodes when they're made private, not if they were already private to begin with?